catalog/ui/search-ui/simple/src/main/java/org/codice/ddf/ui/searchui/simple/properties/UiConfigurationPropertiesFactory.java

```diff
@@ -15,84 +15,48 @@
 
 import java.util.Optional;
 import org.codice.ddf.branding.BrandingRegistry;
+import org.osgi.service.cm.ConfigurationAdmin;
 
 /** Stores external configuration properties. */
-public class ConfigurationStore {
+public class UiConfigurationPropertiesFactory {
 
-  private static ConfigurationStore uniqueInstance;
+  private static final String CONFIG_PID = "ddf.platform.ui.config";
 
-  private String header = "";
+  private static UiConfigurationPropertiesFactory uniqueInstance;
 
-  private String footer = "";
-
-  private String color = "";
+  private Optional<BrandingRegistry> branding = Optional.empty();
 
-  private String background = "";
+  private ConfigurationAdmin configurationAdmin;
 
-  private Optional<BrandingRegistry> branding = Optional.empty();
+  private UiConfigurationPropertiesFactory() {}
 
-  private ConfigurationStore() {
-    header = "";
-    footer = "";
-    color = "";
-    background = "";
+  /**
+   * @return a new UiConfigurationProperties object that contains the properties to build out this
+   *     UI
+   */
+  public UiConfigurationProperties getProperties() {
+    return new UiConfigurationProperties(configurationAdmin, CONFIG_PID, branding);
```

Comment:
❓ I'm wondering if it makes more sense to have this factory class be responsible for querying configAdmin for the configuration and just passing the properties into the `UiConfigurationProperties` object. That would simplify the `UiConfigurationProperties` class and keep the configAdmin dependency in this class only.

Reply:
My original implementation was actually just having Configuration Store (the original class) do the lookup. I'm not actually sure which is better in this case.

Reply:
Agree with @brjeter with getting the `Configuration` object from `ConfigurationAdmin` here. I would let the `UiConfigurationProperties` class take care of getting the values out of the `Configuration` or properties object though. I think it would make for a better separation of concerns.

Reply:
Probably at some point. 
It looks like we created that `PlatformUIConfiguration` to really only be used to get some JSON back for our other UIs so a bit of refactoring would have to be done here.